### PR TITLE
Add Hardware flag to upgrade cmd

### PR DESCRIPTION
**Pull request by chrisdoherty4**
The hardware CSV flag is necessary for upgrade in-case the user wishes to provide the additional Hardware per node group (including control plane). The flag is optional as hardware may already be registered with the cluster that satisfies the upgrade requirements.

A following PR will leverage the input CSV by validating it and cataloging the additional hardware along with existing hardware so we can validate sufficient hardware is available.

### CODE REVIEW

File: cmd/eksctl-anywhere/cmd/tinkerbell.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import (
+	"github.com/spf13/pflag"
+)
+
+// TinkerbellHardwareCSVFlagName is the flag used for providing hardware CSV used by the Tinkerbell
+// provider.
+const TinkerbellHardwareCSVFlagName = "hardware-csv"
+
+// PopulateTinkerbellHardwareCSVFlag populates set with the TinkerbellHardwareCSVFlagName.
```
Comment (chrisdoherty4) [inline cmd/eksctl-anywhere/cmd/tinkerbell.go:11]:
```suggestion
// PopulateTinkerbellHardwareCSVFlag populates s with the TinkerbellHardwareCSVFlagName.
// The value provided to the CLI will be written to field.
```

**Review by abhinavmpandey08 — APPROVED** [review]
/approve

File: cmd/eksctl-anywhere/cmd/upgradecluster.go [hunk 2]
```diff
@@ -58,8 +58,9 @@ func init() {
 	upgradeClusterCmd.Flags().BoolVar(&uc.forceClean, "force-cleanup", false, "Force deletion of previously created bootstrap cluster")
 	upgradeClusterCmd.Flags().StringVar(&uc.bundlesOverride, "bundles-override", "", "Override default Bundles manifest (not recommended)")
 	upgradeClusterCmd.Flags().StringVar(&uc.managementKubeconfig, "kubeconfig", "", "Management cluster kubeconfig file")
-	err := upgradeClusterCmd.MarkFlagRequired("filename")
-	if err != nil {
+	PopulateTinkerbelHardwareCSVFlag(&uc.hardwareCSVPath, upgradeClusterCmd.Flags())
+
```
Comment (ptrivedi) [inline cmd/eksctl-anywhere/cmd/upgradecluster.go:62]: same comment as above. I would like seeing a consistent upgradeClusterCmd.Flags().StringVar call here

File: cmd/eksctl-anywhere/cmd/tinkerbell.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+// TinkerbellHardwareCSVFlagName is the flag used for providing hardware CSV used by the Tinkerbell
+// provider.
+const TinkerbellHardwareCSVFlagName = "hardware-csv"
+
+// PopulateTinkerbellHardwareCSVFlag populates s with the TinkerbellHardwareCSVFlagName.
+// The value provided to the CLI will be written to field.
+func PopulateTinkerbelHardwareCSVFlag(field *string, s *pflag.FlagSet) {
+	s.StringVar(field, TinkerbellHardwareCSVFlagName, "",
+		"A file path to a CSV file containing hardware data to be submitted to the cluster for provisioning.")
+}
```
[the remark this answers is not in the document]
Comment (chrisdoherty4) [reply cmd/eksctl-anywhere/cmd/tinkerbell.go:16]: Could you elaborate on why its convoluted. `PopulateTinkerbellHardwareCSVFlag` seems pretty clear. 

Is this more about not wanting to have a function? We've talked about wanting an alias too so you're proposing 3 constants (flag name, flag alias and flag description) populated, hopefully consistently, in 3 places (generate, update and create) as opposed to 1 function that consistently applies all 3 constants?

In this instance, I value consistency for the end-user vs consistency of code given this is the users interface and a function seems like a pretty sane way to enforce that consistency.

[the remark this answers is not in the document]
Comment (ptrivedi) [reply cmd/eksctl-anywhere/cmd/tinkerbell.go:16]: Seems more intuitive to have the Flags() setting in createcluster.go/upgradecluster.go consistent. Intuitively, I would have just added a getter for the flag value. But yes, let's chat.

**Pull request merged**